conn_http.go

```diff
@@ -475,49 +475,51 @@ func (h *httpConnect) createRequest(ctx context.Context, reader io.Reader, optio
 
 func (h *httpConnect) prepareRequest(ctx context.Context, query string, options *QueryOptions, headers map[string]string) (*http.Request, error) {
 	if options == nil || len(options.external) == 0 {
-		return h.createRequest(ctx, strings.NewReader(query), options, headers)
+		return h.createRequest(ctx, h.url.String(), strings.NewReader(query), options, headers)
 	}
-	payloadBytes, err := h.createRequestPayloadWithExternalTables(ctx, query, options, headers)
+	requestUrl, payloadBytes, err := h.createRequestPayloadWithExternalTables(ctx, query, options, headers)
 	if err != nil {
 		return nil, err
 	}
-	return h.createRequest(ctx, bytes.NewReader(payloadBytes), options, headers)
+	return h.createRequest(ctx, requestUrl, bytes.NewReader(payloadBytes), options, headers)
 }
 
-func (h *httpConnect) createRequestPayloadWithExternalTables(ctx context.Context, query string, options *QueryOptions, headers map[string]string) ([]byte, error) {
+func (h *httpConnect) createRequestPayloadWithExternalTables(ctx context.Context, query string, options *QueryOptions, headers map[string]string) (string, []byte, error) {
```

Comment:
minor: can we return the entire request instead of a few values here? or make this function mutate request struct
```suggestion
func (h *httpConnect) appendExternalTablesToRequest(ctx context.Context, r *http.Request, query string, options *QueryOptions, headers map[string]string) error {
```